netutils/utils.py

```diff
@@ -88,3 +100,37 @@ def jinja2_convenience_function() -> t.Dict[str, t.Callable[..., t.Any]]:
         function_object = getattr(imported_module, function_name)
         result[jinja2_function_name] = function_object
     return result
+
+
+def get_napalm_getters() -> t.Dict[str, t.Dict[str, bool]]:
+    """Utility to return a dictionary of napalm getters based on install napalm version.
+
+    Returns:
+        Keys are OS and values are a dictionary of supported napalm getters.
+
+    Raises:
+        ImportError: If optional dependency Napalm is not installed.
+
+    Example:
+        >>> from netutils.utils import get_napalm_getters
+        >>> napalm_getters = get_napalm_getters()
+        >>> napalm_getters["eos"]["get_arp_table"]
+        >>> True
+        >>> napalm_getters["eos"]["get_ipv6_neighbors_table"]
+        >>> False
+    """
+    if HAS_NAPALM:
```

Comment:
changed this to have the imports inside the function. that way they're only loaded if the function is called.

Reply:
I think the other pattern is more common

Reply:
More common, but this at least will only load when that one function is called, also the nice trade off of doing it within the function is I can mock these imports in the tests without needing to have `napalm` as an testing dependency in the pyproject.toml.

Reply:
Python does not load in a greedy fashion, this module will only load if you call the module anyway, take a look a this:

```
>>> import netutils
>>> dir(netutils)
['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', '__version__']
>>>
```
If this was greedy, or we explicitly set to set in the init file, we would see things methods like `asn`, `vlans`, etc..


Once this change was introduced https://github.com/networktocode/netutils/commit/cb249f088bbfbdec2addf7c6634c8caab7e0e8a2#diff-14c4cb1fd1f636a25adaea989f0875f63e36e3b18ae2d9f83bd7036975fe9509 this would not load automatically 

I am ultimately not sure that not installing the dependencies in the test environment would pass the POLA. That being said, I've not dealt with this situation too much, so could be good to reach out internally to gather ideas.

Reply:
Its easier for testing etc to have napalm as a development dependency. Makes testing much easier as well, now that i've read about POLA I agree that i'd expect this to be part of the test environment dep.